coordinator/src/main/java/org/apache/uniffle/coordinator/SelectStorageStrategy.java

```diff
@@ -17,24 +17,14 @@
 
 package org.apache.uniffle.coordinator;
 
+import java.util.List;
 import java.util.Map;
 
-import org.apache.uniffle.common.RemoteStorageInfo;
+import org.apache.hadoop.fs.Path;
+
 import org.apache.uniffle.coordinator.LowestIOSampleCostSelectStorageStrategy.RankValue;
 
 public interface SelectStorageStrategy {
 
-  RemoteStorageInfo pickRemoteStorage(String appId);
-
-  void incRemoteStorageCounter(String remoteStoragePath);
-
-  void decRemoteStorageCounter(String storagePath);
-
-  void removePathFromCounter(String storagePath);
-
-  Map<String, RemoteStorageInfo> getAvailableRemoteStorageInfo();
-
-  Map<String, RemoteStorageInfo> getAppIdToRemoteStorageInfo();
-
-  Map<String, RankValue> getRemoteStoragePathRankValue();
+  List<Map.Entry<String, RankValue>> sortPathByRankValue(String path, Path testPath, long time, boolean isHealthy);
```

Comment:
Path seems to be a concept of filesystem. It may be not general.

Reply:
Maybe we can support fileSystem at the beginning. If we need to support db, then there may be more things to consider, or we can add a parameter. If it is a file supported by fileSystem, we will turn on this switch.

Reply:
For `S3` , `COS`, it's OK, right?

Reply:
At least We need consider the object Storage.

Reply:
IIRC, Object storage such as s3 also has the concept of Path.

Reply:
But `cos` doesn't seem to have this concept.

Reply:
As I know, the object storage also can be accessed by Hadoop Filesystem Interface (HCFS), like s3/oss/cos. So this is not a problem

But if we want to involve more storages which are not implemented by HCFS or speed up specified api which is only compatible with original storage api, maybe we should implement uniffle's own filesystem, it can refer to Flink.

Please let me if I'm wrong.

Reply:
Thank you very much for your reminder. Although I am not very impressed with this, I have used a similar method to read files in s3.